src/core/dynamic_system.jl

```diff
@@ -63,84 +98,34 @@ Solves the Lindblad Master equation:
 and returns a Vector of observables evaluated at each time step.
 
 **Fields**
-- `H` -- the Hamiltonian operator (of any subtype of `AbstractOperator`).
-- `ψ_0` -- initital state (Ket) of a quantum system
-- `t_range` -- time interval for which the system has to be simulated. 
-        For instance: 
-            t_range=0:10 evaluates the output using time 
-            steps: 0,1,2,...,10. 
-            t_range=0:0.01:1 evaluates the output using 
-            time steps: 0,0.01,0.02,...,1.0 
-- `e_ops` -- list of operators for which the expected value 
-        (the observables) will be evaluated at each time step in t_range. 
-- `c_ops` -- list of collapse operators ``L_i``'s.
+- `problem` -- An object of type LindbladProblem that defines the problem to be solved. 
+- `kwargs` -- list of keyword arguments to be passed to the ODE solver. See (https://docs.sciml.ai/DiffEqDocs/stable/basics/common_solver_opts/#solver_options)..
 """
 function mesolve(
-    H::AbstractOperator, 
-    ρ_0::AbstractOperator, 
-    t::StepRangeLen; 
-    c_ops::Vector{T} where {T<:AbstractOperator}=(T)[], 
-    e_ops::Vector{T} where {T<:AbstractOperator}=(T)[], 
-    )::Matrix{<:Real}
-    Hamiltonian(t) = H
-    mesolve(Hamiltonian, ρ_0, t; c_ops=c_ops, e_ops=e_ops)
-end
+    problem::LindbladProblem;
+    kwargs...
+    )
 
-function mesolve(
-    H::Function, 
-    ρ_0::AbstractOperator, 
-    t::StepRangeLen; 
-    c_ops::Vector{T} where {T<:AbstractOperator}=(T)[], 
-    e_ops::Vector{T} where {T<:AbstractOperator}=(T)[], 
-)::Matrix{<:Real}
-    n_t = length(t)
-    n_o = length(e_ops)
-    observable=zeros(n_t, n_o) 
-
-    if length(c_ops) == 0
-        eigenvalues, eigenvectors = eigen(ρ_0)
-
-        n, _ = size(ρ_0)
-        for i in 1:n
-            eigenvalue = eigenvalues[i]
-            eigenvector = eigenvectors[i, :]
-            
-            @assert imag(eigenvalue) ≈ 0
-            
-            if real(eigenvalue) ≉ 0
-                _, observable_i = sesolve(H, Ket(eigenvector), t, e_ops=e_ops)
-                observable += observable_i * real(eigenvalue)
-            end
-        end
-        return observable
+    if length(problem.c_ops) == 0
```

Comment:
No. It should fail. The user should explicitly use `sesolve`. The function should just do a `single` thing well.

Reply:
What if a user wants to evolve a mixed system without any collapse operators? This would push the work to the user and not be doing what is intuitively expected.

From the user's perspective, the function is solving the problem given and has only one responsibility. How the problem is solved is an implementation detail. 

The micro-detail of how the function is implemented can easily be refactored into `mesolve` and two internal functions, `_lossy_mesolve` and `_lossless_mesolve` if you are worried about the function's complexity.

Reply:
Then they should use the Shrodinger solver. It makes no sense to use the the master equation if there is no collapse operator.

Reply:
As discussed on teams, 

The function name is modelled after QuTiP's `mesolve`. This isn't correct since we aren't dealing with generic master equation solver. We are only dealing with Lindblad equations. So we renamed the function to `lindblad_solve`.